idea/src/org/jetbrains/kotlin/idea/intentions/copyConcatenatedStringToClipboard/ConcatenatedStringGenerator.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.intentions.copyConcatenatedStringToClipboard
+
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.collectDescendantsOfType
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+
+class ConcatenatedStringGenerator {
+    fun create(element: KtBinaryExpression): String {
+        val binaryExpression = getBinaryParentExpression(element)
+        val stringBuilder = StringBuilder()
+        create(binaryExpression, stringBuilder)
+        return stringBuilder.toString()
+    }
+
+    private fun getBinaryParentExpression(element: KtBinaryExpression): KtBinaryExpression {
+        var binaryExpression = element.getStrictParentOfType<KtBinaryExpression>() ?: return element
+        while (binaryExpression != null) {
+            binaryExpression = binaryExpression.getStrictParentOfType<KtBinaryExpression>() ?: return element
+        }
+        return element
+    }
+
+    private fun create(element: KtBinaryExpression, sb: StringBuilder) {
```

Comment:
I'd declare it as something like `private fun KtBinaryExpression.appendTo(sb: StringBuilder)`. And the same for the next function.